File: dp_labs_and_solutions/dp_labs_and_solutions/dp_cpp_labs_and_solutions/command/lab/go_straight_command.cpp
#include "go_straight_command.h"

go_straight_command::go_straight_command(driver* a_driver, int a_how_far) : simple_driver_command(a_driver)
{
	set_how_far(a_how_far);
}

void go_straight_command::execute()
{
	my_driver->go_straight(my_how_far);
}

void go_straight_command::undo()
{
	my_driver->undo_go_straight(my_how_far);
}

void go_straight_command::set_how_far(int a_how_far)
{
	my_how_far = a_how_far;
}

int go_straight_command::get_how_far()
{
	return my_how_far;
}

File: dp_labs_and_solutions/dp_labs_and_solutions/dp_java_labs_and_solutions/command/solution/GoStraightCommand.java
// GoStraightCommand.java

public class GoStraightCommand extends SimpleDriverCommand
{
	public GoStraightCommand(Driver aDriver, int aHowFar)
	{
		super(aDriver);
		setHowFar(aHowFar);
	}

	public void execute()
	{
		getDriver().goStraight(getHowFar());
	}

	public void undo()
	{
		getDriver().undoGoStraight(getHowFar());
	}

	private void setHowFar(int aHowFar)
	{
		myHowFar = aHowFar;
	}

	private int getHowFar()
	{
		return myHowFar;
	}

	private int myHowFar;
}
